src/DynamoCoreWpf/NodeViewCustomization/NodeViewCustomizations/FunctionNodeViewCustomization.cs

```diff
@@ -93,17 +93,31 @@ private void EditCustomNodeProperties()
 
             if (args.Success)
             {
-                CustomNodeWorkspaceModel ws;
-                model.CustomNodeManager.TryGetFunctionWorkspace(
-                    functionNodeModel.Definition.FunctionId,
-                    DynamoModel.IsTestMode,
-                    out ws);
-                ws.SetInfo(args.Name, args.Category, args.Description);
-
-                if (!string.IsNullOrEmpty(ws.FileName))
-                {
-                    ws.Save(ws.FileName, false, dynamoViewModel.EngineController);
-                }
+                SerializeCustomNodeWorkspaceWithNewInfo(args, dynamoViewModel, functionNodeModel);
+            }
+        }
+
+        /// <summary>
+        /// Serialize and update dyf based on FunctionNamePromptEventArgs
+        /// </summary>
+        /// <param name="args">FunctionNamePromptEventArgs which contains updated dyf info</param>
+        /// <param name="dynamoViewModel">Dynamo View Model</param>
+        /// <param name="functionNodeModel">Custom Node</param>
+        internal void SerializeCustomNodeWorkspaceWithNewInfo(FunctionNamePromptEventArgs args, DynamoViewModel dynamoViewModel, Function functionNodeModel)
+        {
+            CustomNodeWorkspaceModel ws;
+            dynamoViewModel.Model.CustomNodeManager.TryGetFunctionWorkspace(
+                functionNodeModel.Definition.FunctionId,
+                DynamoModel.IsTestMode,
+                out ws);
+            ws.SetInfo(args.Name, args.Category, args.Description);
+
+            if (!string.IsNullOrEmpty(ws.FileName))
+            {
+                // Construct a temp WorkspaceViewModel based on for serializing
```

Comment:
can you fix this comment up a bit - grammar seems a bit weird and it doesn't explain to me why this has to happen if I didn't know already.